edx_rest_framework_extensions/tests/test_permission.py

```diff
@@ -27,3 +36,103 @@ def test_has_permission_with_invalid_users(self, user):
         request.user = user
         permission = IsSuperuser()
         self.assertFalse(permission.has_permission(request, None))
+
+
+class JwtHasScopeTests(TestCase):
+    """ Tests for JwtHasScope permission class. """
+    def setUp(self):
+        super(JwtHasScopeTests, self).setUp()
+        self.user = UserFactory()
+        self.scopes = ('grades:read', 'certificates:read')
+
+    @patch('edx_rest_framework_extensions.decorators.waffle.switch_is_active')
+    def test_scopes_validated(self, waffle_mock):
+        """
+        Test that the permission check returns true when scopes are validated.
+        """
+        waffle_mock.return_value = True
+        request = RequestFactory().get('/')
+        request.successful_authenticator = JwtAuthentication()
```

Comment:
Is there an end-to-end test to verify that the `request` object will always have `successful_authenticator` field set as this code expects?

Reply:
Not that I'm aware of. Maybe in the DRF codebase itself? Is that something you think we should add to this package?